### PR TITLE
Implement maybeNetwork & refactor app into functional component and more react style

### PR DESCRIPTION
**Pull request by Jikstra**
- This moves almost all logic from main.js to App.js
- refactors the logic in a react style
- es6 and imports everywhere

Fixes https://github.com/deltachat/deltachat-desktop/issues/1097 & closes https://github.com/deltachat/deltachat-desktop/issues/1095

### CODE REVIEW

File: src/renderer/App.js [hunk 1]
```diff
@@ -1,33 +1,93 @@
...
+  }, [state])
+
+  useEffect(() => {
+    ipcRenderer.on('chooseLanguage', onChooseLanguage)
+    return () => {
+      ipcRenderer.removeListener('chooseLanguage', onChooseLanguage)
+    }
+  }, [localeData])
+
+  if (!localeData) return null
```
Comment (Simon-Laux) [inline src/renderer/App.js]: Is this a temporary loading case? else we should consider displaying an error screen here.

**Comment by Simon-Laux** [comment]
referenced node version seems incompatible:
```
/tmp/.mount_deltactU3O9g/resources/app.asar/src/main/index.js:30
  throw err
  ^

TypeError: this._dc.setStockTranslation is not a function
    at Object.keys.forEach.key (/tmp/.mount_deltactU3O9g/resources/app.asar/src/main/deltachat/login.js:20:16)
    at Array.forEach (<anonymous>)
    at DCLoginController.setCoreStrings (/tmp/.mount_deltactU3O9g/resources/app.asar/src/main/deltachat/login.js:19:26)
    at _dc.open.err (/tmp/.mount_deltactU3O9g/resources/app.asar/src/main/deltachat/login.js:42:12)
    at binding.dcn_open.err (/tmp/.mount_deltactU3O9g/resources/app.asar/node_modules/deltachat-node/lib/deltachat.js:578:9)
Segmentation fault (core dumped)
```

File: src/renderer/App.js (hunk 1 above)
Comment (Jikstra) [reply src/renderer/App.js]: Yup it is. This only happens on the initial loadup.

**Comment by Jikstra** [comment]
> referenced node version seems incompatible:

This looks like your controller cleanup is already incompatible? I didn't change a thing there.

**Pull request merged**